signatures.md

```diff
@@ -52,163 +49,350 @@ Note: `Message` is used here as the object being signed, but this interface shou
 ```go
 type Signature interface {
 
-        // Sign generates a proof that miner `M` generate message `m`
+        // Sign generates a proof that miner `M` generated or approved of message `msg`.
         //
         // Out:
-        //    sig - a series of bytes representing a signature usually `r`|`s`
-        //    err - a standard error message indicating any process issues
+        //   sig - a series of bytes representing a signature  
+        //   err - a standard error message indicating any process issues
         // In:
-        //    m - a series of bytes representing a message to be signed
-        //    sk - a private key which cryptographically links `M` to `sig`
+        //   msg - a series of bytes representing a message to be signed
+  		  //   addr - the Filecoin `Address` of the signer
+        //   sk - a private key which cryptographically links `M` to `sig`
         //
-        Sign(m Message, sk PrivateKey) (sig SignatureBytes, err error)
-
-        // Verify validates the statement: only `M` could have generated `sig`
+        //Sign(msg Message, addr Address, sk PrivateKey) (sig SignatureBytes, err error)
+        
+        // Aggregate generates a BLS signature aggregate from a set of BLS 
+        // signatures. Associated `Address` must be passed to this function
+  		  // so that the aggregate can be verified. This function will pair a `sig` 
+  			// with an `Address` based on index in `sigs` and `addresses`.
+        //
+        // Out:
+        // 	 sig - a series of bytes representing a single BLS signature
+        //   err - a standard error message indicating any process issues
+        // In:
+     		//   sigs - an array of SignatureBytes {sig_1, sig_2..., sig_n}
+    		//   addrs - an array of Addresses {address_sig_1, address_sig_2..., address_sig_n}
+  			//
+  			Aggregate(sigs []SignatureBytes, addrs []Address) (sig SignatureBytes, err error)
+  
+  			// Verify validates the statement: only `M` could have generated `sig`
```

Comment:
whuiii, I think @whyrusleeping wrote a tool to format code blocks in markdown :)

Reply:
Awesome tool. Thanks!